### ISSUE TITLE
[Error] loop argument must agree with lock

### ISSUE THREAD

**Issue by systemime** [aio-libs/aioredis-py#1204]
### Describe the bug

loop argument must agree with lock

### To Reproduce

I push websockeet in django project, but occasionally there is a problem of' loop argument must agree with lock.

In another project, I found a similar problem: [https://github.com/aio-libs/janus/issues/358](https://github.com/aio-libs/janus/issues/358)

### Expected behavior

No exception

### Logs/tracebacks

```python-traceback
aioredis/pool.py in __init__ at line 95
        self._cond = asyncio.Condition(lock=Lock())

asyncio/locks.py in __init__ at line 234
        def __init__(self, lock=None, *, loop=mixins._marker):
                super().__init__(loop=loop)
                if lock is None:
                    lock = Lock()
                elif lock._loop is not self._get_loop():
                 > raise ValueError("loop argument must agree with lock")
```
```


### Python Version

```console
$ python --version

3.10
```

### aioredis Version

```console
$ python -m pip show aioredis

1.3.1
```

### Code of Conduct

- [X] I agree to follow the aio-libs Code of Conduct

**Comment by asvetlov** [comment]
CPython 3.10.0 has a bug regarding the usage of asyncio Condition with explicit lock.

It is fixed on the main branch and will be published as a part of 3.10.1 release.

Right now there is a workaround, like used in [janus](https://github.com/aio-libs/janus/blob/master/janus/__init__.py#L39-L44) project:
```
       self._async_mutex = asyncio.Lock()
        if sys.version_info[:3] == (3, 10, 0):
            # Workaround for Python 3.10 bug, see #358:
            getattr(self._async_mutex, '_get_loop', lambda: None)()
        self._async_not_empty = asyncio.Condition(self._async_mutex)
        self._async_not_full = asyncio.Condition(self._async_mutex)
```

**Comment by systemime** [comment]
> CPython 3.10.0 has a bug regarding the usage of asyncio Condition with explicit lock.
> 
> It is fixed on the main branch and will be published as a part of 3.10.1 release.
> 
> Right now there is a workaround, like used in [janus](https://github.com/aio-libs/janus/blob/master/janus/__init__.py#L39-L44) project:
> 
> ```
>        self._async_mutex = asyncio.Lock()
>         if sys.version_info[:3] == (3, 10, 0):
>             # Workaround for Python 3.10 bug, see #358:
>             getattr(self._async_mutex, '_get_loop', lambda: None)()
>         self._async_not_empty = asyncio.Condition(self._async_mutex)
>         self._async_not_full = asyncio.Condition(self._async_mutex)
> ```

Thanks for the explanation, I will close this issue.
